### PR TITLE
Fixing remaining SYS_LOG references in net

### PR DESCRIPTION
**Pull request by jukkar**

### CODE REVIEW

**Review by jhedberg — APPROVED** [review]
net_buf changes look fine to me

**Comment by jukkar** [comment]
@rlubos / @KaSroka I changed the OpenThread code a bit, does it look ok to you?

File: subsys/net/l2/openthread/openthread_utils.c [hunk 1]
```diff
@@ -114,16 +113,14 @@ void add_ipv6_addr_to_ot(struct openthread_context *context)
 
 	otIp6AddUnicastAddress(context->instance, &addr);
 
-#if CONFIG_OPENTHREAD_L2_LOG_LEVEL == SYS_LOG_LEVEL_DEBUG
-	{
+	if (CONFIG_OPENTHREAD_L2_LOG_LEVEL == LOG_LEVEL_DBG) {
```
Comment (KaSroka) [inline subsys/net/l2/openthread/openthread_utils.c]: Why do you use run time if for this?

File: subsys/net/l2/openthread/openthread_utils.c [hunk 2]
```diff
@@ -70,14 +70,13 @@ void add_ipv6_addr_to_zephyr(struct openthread_context *context)
 
 	for (address = otIp6GetUnicastAddresses(context->instance);
 	     address; address = address->mNext) {
-#if CONFIG_OPENTHREAD_L2_LOG_LEVEL == SYS_LOG_LEVEL_DEBUG
```
Comment (KaSroka) [inline subsys/net/l2/openthread/openthread_utils.c]: Compared to other changes this one is missing `if(...)`

File: subsys/net/l2/openthread/Kconfig [hunk 3]
```diff
@@ -32,57 +32,20 @@ menuconfig OPENTHREAD_DEBUG
...
-	default 1 if OPENTHREAD_LOG_LEVEL_ERROR
-	default 2 if OPENTHREAD_LOG_LEVEL_WARNING
-	default 3 if OPENTHREAD_LOG_LEVEL_INFO
-	default 4 if OPENTHREAD_LOG_LEVEL_DEBUG
-	default 0
+module = OPENTHREAD
+module-dep = NET_LOG
+module-str = Log level for OpenThread stack
+module-help = Enables OpenThread stack code to output debug messages.
+source "subsys/net/Kconfig.template.log_config.net"
```
Comment (KaSroka) [inline subsys/net/l2/openthread/Kconfig]: Will this work as previously implemented map for level to value mapping?

Comment (jukkar) [reply subsys/net/l2/openthread/Kconfig]: Yes, it is a same thing

File: subsys/net/l2/openthread/openthread_utils.c (hunk 1 above)
Comment (jukkar) [reply subsys/net/l2/openthread/openthread_utils.c]: Other option would have been to add `{ char buf[]; ... }` but that looked ugly.

File: subsys/net/l2/openthread/Kconfig (hunk 3 above)
Comment (KaSroka) [reply subsys/net/l2/openthread/Kconfig]: So my only concern is that OpenThread values may diverge from Zephyr values some day.

File: subsys/net/l2/openthread/openthread_utils.c (hunk 1 above)
Comment (jhedberg) [reply subsys/net/l2/openthread/openthread_utils.c]: FWIW, this is not runtime in practice. It's the same thing as IS_ENABLED() (as far as I see), i.e. the branch should get stripped away from the final binary if the condition is not true

Comment (KaSroka) [reply subsys/net/l2/openthread/openthread_utils.c]: Yes it should. Maybe not in -O0, but generally it should be stripped.

File: subsys/net/l2/openthread/Kconfig (hunk 3 above)
Comment (rlubos) [reply subsys/net/l2/openthread/Kconfig]: The problem is that it already diverged (but we do not have this version in Zephyr yet):
https://github.com/openthread/openthread/blob/71d4ad3e777672a441e238344290df302ad86bb7/include/openthread/platform/logging.h#L89

Actually, we should separate OpenThread internal logging configuration from Zephyr logs. I'd say, to keep the OPENTHREAD_LOG_LEVEL as it is, and use it only to configure log level in OpenThread. That shouldn't be difficult, as the only module that uses this config for Zephyr logs which actually prints some logs is `openthread/platform/radio.c` . We could use OPENTHREAD_L2_LOG_LEVEL there instead perhaps?

Comment (jukkar) [reply subsys/net/l2/openthread/Kconfig]: Ok, that makes sense. I will send a new version that restores OPENTHREAD_LOG_LEVEL. I will leave the radio.c changes to you do at later time.

**Review by KaSroka — APPROVED** [review]

**Pull request merged**